code/datums/status_effects/debuffs/staggered.dm

```diff
@@ -43,3 +47,82 @@
 /mob/living/proc/do_stagger_animation()
 	animate(src, pixel_x = 4, time = 0.2 SECONDS, loop = 6, flags = ANIMATION_RELATIVE|ANIMATION_PARALLEL)
 	animate(pixel_x = -4, time = 0.2 SECONDS, flags = ANIMATION_RELATIVE)
+
+/// Status effect specifically for instances where someone is vulnerable to being stunned when shoved.
+/datum/status_effect/next_shove_stuns
+	id = "next shove stuns"
+	duration = 3 SECONDS
+	status_type = STATUS_EFFECT_UNIQUE
+	tick_interval = 0.5 SECONDS
+	alert_type = null
+	remove_on_fullheal = TRUE
+	/// Our visual cue for the vulnerable state this status effect puts us in.
+	var/mutable_appearance/vulnverability_overlay
+
+/datum/status_effect/next_shove_stuns/on_apply()
+	//Let's just clear this if they're dead or we can't stun them on a shove
+	if(owner.stat == DEAD || HAS_TRAIT(owner, TRAIT_NO_SIDE_KICK) || HAS_TRAIT(owner, TRAIT_IMMOBILIZED))
+		return FALSE
+	RegisterSignals(owner, list(COMSIG_LIVING_STATUS_PARALYZE, COMSIG_LIVING_STATUS_STUN), PROC_REF(clear_stun_vulnverability))
+	ADD_TRAIT(owner, TRAIT_STUN_ON_NEXT_SHOVE, STATUS_EFFECT_TRAIT)
+	vulnverability_overlay = mutable_appearance(icon = 'icons/effects/effects.dmi', icon_state = "dazed")
+	owner.add_overlay(vulnverability_overlay)
+	return TRUE
+
+/datum/status_effect/next_shove_stuns/on_remove()
+	UnregisterSignal(owner, list(
+			COMSIG_LIVING_STATUS_PARALYZE,
+			COMSIG_LIVING_STATUS_STUN,
+	))
+	REMOVE_TRAIT(owner, TRAIT_STUN_ON_NEXT_SHOVE, STATUS_EFFECT_TRAIT)
+	if(vulnverability_overlay)
+		clear_stun_vulnverability_overlay()
+
+/datum/status_effect/next_shove_stuns/tick(seconds_between_ticks)
+	// If dead or we can't stun them with a shove, remove this status effect
+	if(owner.stat == DEAD || HAS_TRAIT(owner, TRAIT_NO_SIDE_KICK) || HAS_TRAIT(owner, TRAIT_IMMOBILIZED))
+		clear_stun_vulnverability_overlay()
+		qdel(src)
+		return
```

Comment:
Maybe we should be checking this stuff in the shove stun too, if we're not already? Sounds possible for there to be a race condition where you shove someone over before the tick deletes the shove stun status effect.

Reply:
That's why I had it clear the status using signals as well for that possibility. Having it delete on tick is probably not necessary, but better to be doubly sure. I might also have it signal to clear on death and immobilized as well just be sure.

Reply:
Let's avoid this code if we have it on signals then because I think that it just makes it easier for someone to add it here and not the other

Reply:
Sounds good to me